libp2p/utils/semaphore.nim

```diff
@@ -19,7 +19,7 @@ type
   AsyncSemaphore* = ref object of RootObj
     size*: int
     count*: int
-    queue*: seq[Future[void]]
+    queue: seq[Future[void]]
 
 proc init*(T: type AsyncSemaphore, size: int): T =
```

Comment:
this should be called `new` because AsyncSemaphore is a `ref object` - alternatively, it should be called `newAsyncSemaphore` because that's what all chronos primitives are called

Reply:
Err, I'm a bit confused with the moving target of constructor naming, you've always pushed for `init` as the end all constructor/initializer name why the sudden change to `new`? 

(btw, I understand the consistency argument and that's fine, but still you just changed the rules :-) ?)

Reply:
`init` for `object`, `new` for `ref object` - no change here - it's just that I almost never use `ref object`, they're semantically problematic

Reply:
Everything in libp2p is a `ref object` and yet we've had numerous discussions to change it to init. OK, I'll remember to use that convention from now on.

Reply:
these discussions have generally been preceded by a discussion to turn some refs into plain objects - ie it's always possible to create a `ref` from an `object` - the other way around is not possible - so generally, unless there are ownership reasons (which _usually_ are false), it's better to use `object` and explicitly say `ref Xxx` where necessary

Reply:
Dynamic dispatch requires `ref` objects, even if using an ad-hoc dispatch table, that why libp2p uses `ref` for the most part.